prospero-cli/src/main/java/org/wildfly/prospero/cli/RepositoryDefinition.java

```diff
@@ -17,43 +17,93 @@
-            final String text = repos.get(i);
+            String text = repos.get(i);
             if(text.contains("::")) {
                 final String[] parts = text.split("::");
                 if (parts.length != 2 || parts[0].isEmpty() || parts[1].isEmpty() || !isValidUrl(parts[1])) {
                     throw CliMessages.MESSAGES.invalidRepositoryDefinition(text);
                 }
-
                 repositories.add(new Repository(parts[0], parts[1]));
             } else {
+                if (!text.startsWith("http") && !text.isEmpty()) {
+                    text = getAbsoluteFileURL(text);
+                }
                 if (!isValidUrl(text)) {
                     throw CliMessages.MESSAGES.invalidRepositoryDefinition(text);
                 }
-
                 repositories.add(new Repository("temp-repo-" + i, text));
             }
         }
         return repositories;
     }
 
-    private static boolean isValidUrl(String text) {
+    public static String getAbsoluteFileURL(String text) throws MalformedURLException, URISyntaxException, ArgumentParsingException {
+        Path path = getfilePath(text);
+        if (!path.isAbsolute()) {
+            text = text.replace(path.toString(), "") + Paths.get(path.toString()).toAbsolutePath().normalize();
+        }
+        if (!text.startsWith("file")) {
+            text = "file:" + text;
+        }
+        return new URL(text).toString();
+    }
+
+    private static boolean isValidUrl(String text) throws ArgumentParsingException {
+        if (text.startsWith("file://..")) {
+            Logger.getLogger("'file://../path/to/repo' is invalid as the format of file URL should be " +
+                    "'file://<HOST>/<PATH>'");
+            return false;
+        }
         try {
             new URL(text);
             return true;
-        } catch (MalformedURLException e) {
+        } catch (InvalidPathException | MalformedURLException e) {
             return false;
         }
     }
+
+    private static Path getfilePath(String text) throws MalformedURLException, URISyntaxException, ArgumentParsingException {
+        Path path;
+        if(text.startsWith("file")){
+            path = getPathFromURL(text);
+        } else {
+            path = Paths.get(text);
+        }
+        if (Files.exists(path)) {
+            return path;
+        } else {
+            throw CliMessages.MESSAGES.invalidFilePath(path.toString());
+        }
+    }
+
+    /* Pattern command, exists in order to recognize Windows file paths properly.
+       For Windows systems, Path.of(Paths.get(url.toURI()).toString()) works correctly,
+       whereas for Linux, Paths.get(url.getPath()).toAbsolutePath() is more suitable. */
+    private static Path getPathFromURL(String text) throws MalformedURLException, URISyntaxException, ArgumentParsingException {
+        URL url = new URL(text);
+        if (Pattern.compile("/[A-Z]:").matcher(text).find()) {
```

Comment:
That's not going to correct, e.g. in Linux it's possible to create a folder with name `A:`